var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -19,7 +19,7 @@ class Hip(CMakePackage):
     url      = "https://github.com/ROCm-Developer-Tools/HIP/archive/rocm-4.5.0.tar.gz"
 
     maintainers = ['srekolam', 'arjun-raj-kuppala', 'haampie']
-    version('master', branch='master')
+    version('master', branch='develop')
```

Comment:
Ok. Now I understand a little better why you're choosing these branches. The HIP develop branch is basically the [**staging**](https://github.com/spack/spack/pull/28500#issuecomment-1040694454) concept I described before. (The HIP mainline branch is not publicly available.)

In the math libraries, the develop branch is also basically equivalent to staging. It's nice that we have some consistency about that! And the mathlibs master branch corresponds to mainline. This actually matches with the [spack version comparison](https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison); the develop/staging branch is the newest version, and master/mainline is slightly older than develop/staging but newer than the most recent release.

So, I would argue that rather than naming develop/staging as `master` in spack, we should be naming it as `develop`. For tools and libraries that do have a public mainline branch, we can call that one `master` in spack.